content/en/guides/directory-structure/nuxt-config.md

```diff
@@ -100,16 +100,85 @@ See more on the [dev property](/guides/configuration-glossary/configuration-dev
 
 ### env
 
-This option lets you define environment variables that are available to both client and server.
+This option lets you define environment variables that are required at build time rather than runtime such as `NODE_ENV=staging` or `VERSION=1.2.3`. However for runtime environment variables the runtime config` is preferred.
 
 ```js{}[nuxt.config.js]
 export default {
   env: {
-    baseUrl: process.env.BASE_URL || baseUrl
+    baseUrl: process.env.BASE_URL
   }
 }
 ```
 
+### runtimeConfig
+
+The runtime config has built-in dotenv support providing better security and faster development. The runtime config is added to the Nuxt payload so there is no need to rebuild in order to update the runtime configuration when working in development or with Server-side rendering or client side only applications. Although for static sites you will still need to regenerate your site to see these changes.
+
+#### `publicRuntimeConfig`
+
+- should hold all env variables that are public as these will be exposed on the frontend. This could include a reference to your public URL for example.
+- is available using `$config` in both server and client.
+
+```js{}[nuxt.config.js]
+export default {
+  publicRuntimeConfig: {
+    baseURL: process.env.BASE_URL || 'https://nuxtjs.org'
+  }
+}
+```
+
+#### `privateRuntimeConfig`
+
+- should hold all env variables that are private and that should not be exposed on the frontend. This could include a reference to your API secret tokens for example.
+- is only available on server using same `$config` (it overrides publicRuntimeConfig)
+
+```js{}[nuxt.config.js]
+export default {
+  privateRuntimeConfig: {
+    apiSecret: process.env.API_SECRET
+  }
+}
+```
+
+#### **Using your config values:**
+
+You can then access these values anywhere by using the context in your pages, store, components and plugins by using `this.$config` or `context.$config`.
+
+```html{}[pages/index.vue]
+<script>
+  asyncData ({ $config: { baseURL } }) {
+    const posts = await fetch(`${baseURL}/posts`)
+      .then(res => res.json())
+  }
+</script>
+```
+
+Inside your templates you can access your runtimeConfigs directly using `$config.*`
+
+```html{}[pages/index.vue]
+<template>
+  <p>Our Url is: {{ $config.baseURL}}</p>
+</template>
+```
+
+<base-alert type="warn">
+
+You can expose server/private tokens if using `$config` in fetch/asyncData or directly inside your template.
```

Comment:
```suggestion
It is possible that private config to be exposed to client when using `$config` in `fetch`, `asyncData` or directly inside templates. 
```

/cc @danielroe @manniL  if this can be reworded better

Reply:
```suggestion
Your private config could be exposed if you use `$config` outside of a server-only context (for example, if you use `$config` in `fetch`, `asyncData` or directly inside your template).
```